packages/router/router.ts

```diff
@@ -1765,37 +1775,110 @@ export function unstable_createStaticHandler(
 
   let dataRoutes = convertRoutesToDataRoutes(routes);
 
+  /**
+   * The query() method is intended for document requests, in which we want to
+   * call an optional action and potentially multiple loaders for all nested
+   * routes.  It returns a StaticHandlerContext object, which is very similar
+   * to the router state (location, loaderData, actionData, errors, etc.) and
+   * also adds SSR-specific information such as the statusCode and headers
+   * from action/loaders Responses.
+   *
+   * It _should_ never throw and should report all errors through the
+   * returned context.errors object, properly associating errors to their error
+   * boundary.  Additionally, it tracks _deepestRenderedBoundaryId which can be
+   * used to emulate React error boundaries during SSr by performing a second
+   * pass only down to the boundaryId.
+   *
+   * The one exception where we do not return a StaticHandlerContext is when a
+   * redirect response is returned or thrown from any action/loader.  We
+   * propagate that out and return the raw Response so the HTTP server can
+   * return it directly.
+   */
```

Comment:
@jacob-ebey Let me know if this and the comment above `queryRoute` clarify the behavior (and match what we discussed yesterday).  

The one thing that's slightly different I think is that while in the remix use-case, `callRouteAction` for example will never throw a user-thrown error - this remix-agnostic code will still intend to throw user-thrown things inside of `queryRoute`.  We just shouldn't hit that path in the Remix usage of `remix-run/router`